code/divide_conquer/src/search_in_a_rotated_array/search_in_a_rotated_array.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+Problem statement:
+Given a sorted and rotated array A of N distinct elements which is rotated
+at some point, and given an element K. The task is to find the index of
+the given element K in the array A.
+*/
+
+#include <iostream>
+#define ll long long int
+
+ll findPivot(ll arr[], ll low, ll high) {
+    while (low < high) {
+        // find mid element
+        ll mid = low + (high - low) / 2;
+
+        if (arr[mid] < arr[0]) {
+            high = mid;
+        } else {
+            low = mid + 1;
+        }
+    }
+
+    if (arr[low] < arr[0]) {
+        return low;
+    } else {
+        // pivot element does not exist
+        return -1;
+    }
+}
+
+ll findAns(ll arr[], ll k, ll low, ll high) {
+
+    while (low < high) {
+        // find mid element
+        ll mid = low + (high - low) / 2;
+
+        if (arr[mid] == k) {
+            return mid;
+        }
+
+        if (arr[mid] >= k) {
+            high = mid;
+        } else {
+            low = mid + 1;
+        }
+    }
+
+    if (arr[low] == k) {
+        return low;
+    } else {
+        // element does not exist in array
+        return -1;
+    }
+}
+
+int main() {
+    ll test, n, k, pivot;
+    std::cin >> test; // no. of test cases
+
+    while (test--) {
+        std::cin >> n; // size of array
+        ll arr[n];

```

Comment:
One last thing. This is invalid code because it is not standards compliant. Instead use `std::vector`. If you want to not change much, use a dynamic array instead.

Reply:
Okay, sir, I will change that.